File: day1/main.py
def part1(file):
    left_list, right_list = [], []
    with open(file) as f:
        lines = f.readlines()

    for line in lines:
        split_line = line.split()
        left_list.append(int(split_line[0]))
        right_list.append(int(split_line[1]))

    left_list.sort()
    right_list.sort()

    total = 0
    for i in range(len(left_list)):
        total += abs(left_list[i] - right_list[i])
    return total


def part2(file):
    left_list, right_list = [], []
    with open(file) as f:
        lines = f.readlines()

    for line in lines:
        split_line = line.split()
        left_list.append(int(split_line[0]))
        right_list.append(int(split_line[1]))

    total = 0
    for i in left_list:
        occurrences = 0
        for j in right_list:
            if i == j:
                occurrences += 1
        total += i * occurrences

    return total


print(part2("input.txt"))

File: day1/main.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	fmt.Println(partOne("input.txt"))
}

func partOne(filename string) int {
	file, err := os.Open(filename)
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	var first, second []int

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		nums := strings.Fields(line)
		num_one, _ := strconv.Atoi(nums[0])
		num_two, _ := strconv.Atoi(nums[1])

		first = append(first, num_one)
		second = append(second, num_two)
	}

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}

	sort.Ints(first)
	sort.Ints(second)

	total := 0

	for i, _ := range first {
		diff := abs(second[i] - first[i])
		total += diff
	}

	return total
}

func abs(num int) int {
	if num < 0 {
		return -num
	} else {
		return num
	}
}
